### ISSUE TITLE
Gulp not live-reloading

### ISSUE THREAD

**Issue by ongmin** [coryhouse/react-flux-starter-kit#4]
Hello Cory, 

took your course on pluralsight. My livereload on gulp isn't running and I've have to stop and start on CLI to rebuild whatever changes I've been making. Any idea why? 

Thanks in advance.

ps. Another student had this problem [link](https://github.com/AveVlad/gulp-connect/issues/153)

```
'use strict'

var gulp = require('gulp')
var connect = require('gulp-connect') // Runs a local dev server
var open = require('gulp-open') // Open a URL in a web browser
var browserify = require('browserify') // Bundles JS
var reactify = require('reactify')  // Transforms React JSX to JS
var source = require('vinyl-source-stream') // Use conventional text streams with Gulp
var concat = require('gulp-concat') // Concatenates files

var config = {
  port: process.env.PORT || 9005,
  devBaseUrl: 'http://localhost',
  paths: {
    html: './src/*.html',
    js: [
      './src/main.js',
      './src/js/*.js'
    ],
    images: './src/images/*',
    css: [
      'node_modules/bootstrap/dist/css/bootstrap.min.css',
      'node_modules/bootstrap/dist/css/bootstrap-theme.min.css',
      'node_modules/toastr/toastr.css',
      './src/css/*.css'
    ],
    dist: './dist',
    mainJs: './src/main.js'
  }
}

// Start a local development server
gulp.task('connect', function () {
  connect.server({
    root: ['dist'],
    port: config.port,
    base: config.devBaseUrl,
    livereload: true
  })
})

gulp.task('open', ['connect'], function () {
  gulp.src('dist/index.html')
    .pipe(open({ uri: config.devBaseUrl + ':' + config.port + '/' }))
})

gulp.task('html', function () {
  gulp.src(config.paths.html)
    .pipe(gulp.dest(config.paths.dist))
    .pipe(connect.reload())
})

gulp.task('js', function () {
  browserify(config.paths.mainJs)
    .transform(reactify)
    .bundle()
    .on('error', console.error.bind(console))
    .pipe(source('bundle.js'))
    .pipe(gulp.dest(config.paths.dist + '/scripts'))
    .pipe(connect.reload())
})

gulp.task('css', function () {
  gulp.src(config.paths.css)
    .pipe(concat('bundle.css'))
    .pipe(gulp.dest(config.paths.dist + '/css'))
})

// Migrates images to dist folder
gulp.task('images', function () {
  gulp.src(config.paths.images)
    .pipe(gulp.dest(config.paths.dist + '/images'))
    .pipe(connect.reload())

  // publish favicon
  gulp.src('./src/favicon.ico')
        .pipe(gulp.dest(config.paths.dist))
})

gulp.task('watch', function () {
  gulp.watch(config.paths.html, ['html'])
  gulp.watch(config.paths.js, ['js'])
  gulp.watch(config.paths.css, ['css'])
})

gulp.task('default', ['html', 'js', 'css', 'images', 'open', 'watch'])

```

**Comment by coryhouse** [comment]
Hi, can you please post this on the course discussion so others can benefit?

Thanks!
Cory

**Comment by ongmin** [comment]
Unfortunately I've closed my pluralsight account. I did your course last December, but decided to reuse the base to build something else. Not sure I can post on pluralsight again without the monthly fee.

Sorry about that. If you ever post a solution, do let me know.
Thank you again.

Min

**Comment by johnboy14** [comment]
I have the same problem also.  Be aware I am using Chromium on Ubuntu to take this course, but I assume that should not be a problem.

**Comment by coryhouse** [comment]
@johnboy14 I'd suggest trying to upgrade gulp-connect. Could also post an issue on that repo.

**Comment by manideepcs024** [comment]
try shifting watch task as first task rather as last task :+1: 
gulp.task('default', [ 'watch', 'html', 'js', 'lint', 'css', 'open']);
 Worked in my case

**Comment by asierba** [comment]
I was having a similar issue. 
After a while I realised that somehow I was missing the livereload.js script in my html page.
I added manually and it worked: 
`<script src="//localhost:35729/livereload.js?snipver=1" async="" defer=""></script>`

Actually, the real problem was that my html page didn't have a proper html format. I didnt even have the html tags on it - just a a file with text. So I guess that's why the html wasn't inserted in the page - the scripts gets inserted just before the closing `</html>` tag! Silly me.. 

Hope this helps
